tests/lac/schur_complement_01_instrumented_vmult.cc

```diff
@@ -0,0 +1,276 @@
+#include <deal.II/lac/schur_complement.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include "../tests.h"
+
+#define PRINTME(name, var) \
+  deallog << "Solution vector: " << name << ": " << var << std::endl;
+
+
+
+int
+main()
+{
+  initlog();
+  deallog.depth_console(0);
+  deallog << std::setprecision(10);
+
+  // deal.II SparseMatrix
+  {
+    deallog << "Schur complement" << std::endl;
+    deallog.push("SC_SparseMatrix");
+
+    {
+      deallog << "SparseMatrix 1" << std::endl;
+
+      /* MATLAB / Gnu Octave code
+
+        clear all;
+        printf("SparseMatrix 1")
+        A = [1,2;3,4]
+        b = [5;6]
+        x = A\b
+
+       */
+
+      const unsigned int rc = 1;
+      SparsityPattern    sparsity_pattern(rc, rc, 0);
+      sparsity_pattern.compress();
+
+      SparseMatrix<double> A(sparsity_pattern);
+      SparseMatrix<double> B(sparsity_pattern);
+      SparseMatrix<double> C(sparsity_pattern);
+      SparseMatrix<double> D(sparsity_pattern);
+      Vector<double>       x(rc);
+      Vector<double>       y(rc);
+      Vector<double>       f(rc);
+      Vector<double>       g(rc);
+      for (unsigned int i = 0; i < rc; ++i)
+        {
+          A.diag_element(i) = 1.0 * (i + 1);
+          B.diag_element(i) = 2.0 * (i + 1);
+          C.diag_element(i) = 3.0 * (i + 1);
+          D.diag_element(i) = 4.0 * (i + 1);
+          f(i)              = 5.0 * (i + 1);
+          g(i)              = 6.0 * (i + 1);
+        }
+
+      const auto lo_A = linear_operator(A);
+      const auto lo_B = linear_operator(B);
+      const auto lo_C = linear_operator(C);
+      const auto lo_D = linear_operator(D);
+
+      SolverControl            solver_control_A(1, 1.0e-10, false, false);
+      SolverCG<Vector<double>> solver_A(solver_control_A);
+      PreconditionJacobi<SparseMatrix<double>> preconditioner_A;
+      preconditioner_A.initialize(A);
+      auto lo_A_inv = inverse_operator(lo_A, solver_A, preconditioner_A);
+      lo_A_inv.vmult = [base_vmult = lo_A_inv.vmult](Vector<double> &dst,
+                                                     const Vector<double> &src) 
+        {
+          deallog << "Calling lo_A_inv.vmult()" << std::endl;
+          base_vmult(dst, src);
+        };
+
+      const auto lo_S = schur_complement(lo_A_inv, lo_B, lo_C, lo_D);
+
+      SolverControl            solver_control_S(1, 1.0e-10, false, false);
+      SolverCG<Vector<double>> solver_S(solver_control_S);
+      PreconditionJacobi<SparseMatrix<double>> preconditioner_S;
+      preconditioner_S.initialize(D); // Same space as S
+      auto lo_S_inv = inverse_operator(lo_S, solver_S, preconditioner_S);
+      lo_S_inv.vmult = [base_vmult = lo_S_inv.vmult](Vector<double> &dst,
+                                                     const Vector<double> &src) 
+        {
+          deallog << "Calling lo_S_inv.vmult()" << std::endl;
+          base_vmult(dst, src);
+        };
```

Comment:
This and the corresponding blocks below are the additions to the test, compared to `schur_complement_01.cc`.